### DISCUSSION TITLE
Supabase + Nextjs app - working with supabase's auth component

### DISCUSSION THREAD

**Discussion by herropaul** [supabase/supabase#11876] [Questions]
Hey supabase community,

I'm not sure whether this question falls under the supabase or nextjs category but I have some problems working with how the Auth component from Supabase works.

Problem: I am using the Auth component from Supabase to authenticate users. After logging in, is there a way to redirect the user to a dynamic route such as `localhost:3000/[username]` where I can pass in the user's data from supabase?

My current work right now is conditionally rendering a separate component by checking to see if a session exists or not from authenticating. Below is what I currently have: 
<img width="478" alt="Screen Shot 2023-01-23 at 2 10 00 PM" src="https://user-images.githubusercontent.com/35973829/214161150-e5211967-0575-4a6d-b438-d3af2e49f7d1.png">

**Comment by silentworks** [comment]
Hey Paul, thanks for asking the question here. Is the login just a email/password login or is it using social auth?

**Reply by herropaul** [reply]
Hey Andrew! Thanks for getting back to me. And this is an email/password login im using. Is there a prop that I'm not aware of for the `Auth` component that'll be able to redirect me to a given path?

**Reply by silentworks** [reply]
Hey Paul, when using email/password login you would have to handle redirect yourself in your app, the component wouldn't do this for you. In the case of using NextJS I would say it's probably better to just create your own form for now as the `auth-ui` stuff doesn't work well in a SSR environment, we are working to improve this but don't have a version out as yet that handles this well.
